Fixing Cluster Settings layout

aleksfront opened the pull request:
Fixing overall layout issues as well as other bugs with a bit of refactoring (mostly removing extra input logic - it can be added later to the `Input` component itself later on).

Fixing bugs listed on https://github.com/lensapp/lens/pull/585#issuecomment-667994826 (except of Icon uploader bugs)

![settings](https://user-images.githubusercontent.com/9607060/89520611-c8d3e000-d7e6-11ea-9f7e-d43d579626b9.gif)

aleksfront committed:
A bit of cleaning in Add Cluster page

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Adding head-col to WizardLayout

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Fixing Cluster Settings general layout bugs

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Cluster Status view refactoring

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Install Metrics component refactoring

Using notifications for error, removed picking
button icon method, simplified button generation.

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Remove icons / checks from RemoveClusterButton

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Fixing colorError in Input styles

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Preventing Input's spellchecking

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
ClusterNameSettings refactoring

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
ClusterWorkspaceSettings refactoring/fixing

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
ClusterProxySetting refactoring

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
ClusterPrometheusSetting refactoring

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Clean up Removal section

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Glued InstallMetrics & InstallUserMode into 1 component

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Removing unused styles in Cluster Settings

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Merge branch 'vue_react_migration' into fix/cluster-settings-layout

aleksfront committed:
Cluster Settings styling

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Adding close button to settings header

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
ClusterHomeDirSetting refactoring

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
FilePicker restyling

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Fixing Prometheus selector

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Fixing Hashicon

Passing cluster name instead of cluster id to prevent
icon changing while typing new
cluster name

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

ixrock commented on src/renderer/components/+cluster-settings/components/install-feature.tsx:
```diff
@@ -0,0 +1,81 @@
+    );
+  }
+
+  runAction(action: keyof typeof clusterIpc): () => Promise<void> {
```
I would use callback here instead of fixed ipc-actions list.

ixrock commented on src/renderer/components/+cluster-settings/components/remove-cluster-button.tsx:
```diff
@@ -1,63 +1,38 @@
-          this.status = RemovalStatus.ERROR;
-          this.errorText = err.toString();
-        }
+        await clusterIpc.disconnect.invokeFromRenderer(cluster.id);
```
Disconnect before removing probably won't needed, plz try to remove active cluster and see for any errors.
P.S. in `clusters-menu.tsx` only `clusterStore.removeById(cluster.id)` is used to remove cluster.

ixrock commented on src/renderer/components/file-picker/file-picker.tsx:
```diff
@@ -43,7 +43,7 @@ export enum OverTotalSizeLimitStyle {
 export interface BaseProps {
     accept?: string;
-    labelText: string;
+    label: React.ReactElement;
```
Better to use `React.ReactNode` as data-layout placeholder

ixrock reviewed (approved):
LGTM

aleksfront committed:
Minor ClusterSettings fixes

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

Nokel81 commented on src/renderer/components/+cluster-settings/cluster-settings.tsx:
```diff
@@ -1,25 +1,43 @@
-    
+    const header = (
+      <>
+        <ClusterIcon
```
It looks weird to have this icon be at 55% opacity

Nokel81 commented on src/renderer/components/+cluster-settings/components/cluster-home-dir-setting.tsx:
```diff
@@ -1,86 +1,44 @@
-    this.directory = directory;
+  onChange = (value: string) => {
+    this.directory = value;
+    this.save(value);
```
If something is happening in the background and looks up the working directory value while it is being typed in, wouldn't it get a bad value if we update it on every value change?

aleksfront replied:
Yes, it can happen. In theory, if updating would happened rarely (for example if we construct some `Save` button) same situation can appear in case where user just saved wrong value. @nevalla Wdyt? Is there any background process can be harmed by quick parameter update? Now it updates itself every 500ms while typing.

nevalla replied:
Can we save when the field loses the focus (`onblur`)? Or did I understand the problem wrong?

aleksfront replied:
Sure, that would be best decision 👍

Nokel81 replied:
Why not also on submit (enter) as well?

aleksfront committed:
Increasing opacity for non-interactive icons

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Keep feature install loading state

Waiting for props to change before
disabling loading state (gray button
width spinner)

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Remove arrays in disposeOnUnmount()

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Fix Cluster select behavior

Now clicking cluster icon in sidebar
always leads to / dashboard. And
'Settings' submenu switches active
cluster at first and only the showing
Cluster Settings

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Using structuralComparator in feature installer

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Saving input fields on blur

Signed-off-by: alexfront <alex.andreev.email@gmail.com>

aleksfront committed:
Setting Select color same as Input color

Signed-off-by: alexfront <alex.andreev.email@gmail.com>
